src/DIRAC/Resources/IdProvider/IdProvider.py

```diff
@@ -13,16 +16,20 @@ def __init__(self, **kwargs):
         meta = self.DEFAULT_METADATA
         meta.update(kwargs)
         self.setParameters(meta)
-        self._initialization(**meta)
+        self._initialize()
 
-    def _initialization(self, **kwargs):
+    def _initialize(self):
         """Initialization"""
-        pass
+        self.name = self.parameters.get("ProviderName")
 
-    def setParameters(self, parameters):
+    def setParameters(self, parameters: dict):
```

Comment:
`dict` isn't a valid type hint. The types of the keys should be included in square brackets.

Reply:
Never used it before. Should we probably discuss how to use type hints in the DIRAC code ? Use it systematically, or just when updating codes, or... For example, the syntax without square brackets is apparently also valid as the tests pass.

Reply:
There is an issue for this: https://github.com/DIRACGrid/DIRAC/issues/6315

I think that as for other cases of syntax updates we should rely on existing tools rather than try to do it ourselves.

Reply:
> For example, the syntax without square brackets is apparently also valid as the tests pass.

cpython doesn't use the type hints at runtime and we don't have any CI jobs to check them so you can put almost anything without breaking the tests.